workflow/usage.rst

```diff
@@ -533,3 +542,224 @@ You can access the message from a Twig template as follows:
+                                'max_num_of_words' => 500,
+                            ],
+                        ],
+                        // ...
+                    ],
+                    'transitions' => [
+                        'to_review' => [
+                            'from' => 'draft',
+                            'to' => 'review',
+                            'metadata' => [
+                                'priority' => 0.5,
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ]);
+
+Then you can access this metadata in your controller as follows::
+
+    public function myControllerAction(Registry $registry, Article $article)
+    {
+        $workflow = $registry->get($article);
+
+        $workflow
+            ->getMetadataStore()
+            ->getWorkflowMetadata()['title'] ?? false
+        ;
+
+        // or
+        $workflow->getMetadataStore()
+            ->getWorkflowMetadata()['title'] ?? false
+        ;
+
+        // or
+        $aTransition = $workflow->getDefinition()->getTransitions()[0];
+        $workflow
+            ->getMetadataStore()
+            ->getTransitionMetadata($aTransition)['title'] ?? false
+        ;
+    }
+
+There is a shortcut that works with everything::
+
+    $workflow
+        ->getMetadataStore()
+        ->getMetadata('title')
+    ;
+
+In a Flash message in your Controller::
+
+    // $transition = ...; (an instance of Transition)
+
+    // $workflow is a Workflow instance retrieved from the Registry (see above)
+    $title = $workflow->getMetadataStore()->getMetadata('title', $transition);
+    $this->addFlash('info', "You have successfully applied the transition with title: '$title'");
+
+Metadata can also be accessed in a Listener, from the Event object.
+
+Using transition blockers you can
+return a user-friendly error message when you stop a transition from happening. In the example we
+get this message from the :class:`Symfony\\Component\\Workflow\\Event\\Event`'s metadata, giving
+you a central place to manage the text.
+
+.. tip::
+
+    This example has been simplified; in production you may prefer to use the :doc:`Translation </components/translation>`
+    component to manage messages in one place::
+
+        namespace App\Listener\Workflow\Task;
+
+        use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+        use Symfony\Component\Workflow\Event\GuardEvent;
+        use Symfony\Component\Workflow\TransitionBlocker;
+
+        class OverdueGuard implements EventSubscriberInterface
+        {
+            public function guardPublish(GuardEvent $event)
+            {
+                $timeLimit = $event->getMetadata('time_limit', $event->getTransition());
+
+                if (date('Hi') <= $timeLimit) {
+                    return;
+                }
+
+                $explanation = $event->getMetadata('explanation', $event->getTransition());
```

Comment:
i think you should add an `explanation` entry in the metadata of this publish transition so that a newcomer can directly spot the usage :)